x-pack/plugins/logstash/public/services/license/logstash_license_service.js

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export class LogstashLicenseService {
+  constructor(license, navigateToApp, toasts) {
+    this.license = license;
+    this.navigateToApp = navigateToApp;
+    this.toasts = toasts;
+  }
+
+  get enableLinks() {
+    return this.calculated.enableLinks;
+  }
+
+  get isAvailable() {
+    return this.calculated.isAvailable;
+  }
+
+  get isReadOnly() {
+    return this.calculated.isReadOnly;
+  }
+
+  get message() {
+    return this.calculated.message;
+  }
+
+  get isSecurityEnabled() {
+    return this.license.getFeature(`security`).isEnabled;
+  }
+
+  /**
+   * Checks if the license is valid or the license can perform downgraded UI tasks.
+   * Rejects if the plugin is not available due to license.
+   */
+  checkValidity() {
+    return new Promise((resolve, reject) => {
```

Comment:
It's one of the weirdest ways to check a boolean value I've ever seen.

Reply:
Yeah maybe I'm err'ing too much on the side of "don't changing anything in this plugin that we don't have to"... Just worried about lack of unit test coverage.